federation/pkg/dnsprovider/dns_test.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dnsprovider
+
+import (
+	"testing"
+
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/rrstype"
+)
+
+// Compile time interface check
+var _ ResourceRecordSet = record{}
+
+type record struct {
+	Name_    string
+	Rrdatas_ []string
+	Ttl_     int64
+	Type_    string
+}
+
+func (r record) Name() string {
+	return r.Name_
+}
+
+func (r record) Ttl() int64 {
+	return r.Ttl_
+}
+
+func (r record) Rrdatas() []string {
+	return r.Rrdatas_
+}
+
+func (r record) Type() rrstype.RrsType {
+	return rrstype.RrsType(r.Type_)
+}
+
+const testDNSZone string = "foo.com"
+
+var testData = []struct {
+	inputs         [2]record
+	expectedOutput bool
+}{
+	{
+		[2]record{
+			{"foo", []string{"1.2.3.4", "5,6,7,8"}, 180, "A"}, // Identical
```

Comment:
nit: Commas are kinda weird here, right?

also might be nice to have a test to show that order matters in rrdata (which it should, I believe)